src/usb_cam.cpp

```diff
@@ -82,7 +82,8 @@ void UsbCam::process_image(const char * src, char * & dest, const int & bytes_us
   // TODO(flynneva): could we skip the copy here somehow?
   // If no conversion required, just copy the image from V4L2 buffer
   if (m_image.pixel_format->requires_conversion() == false) {
-    memcpy(dest, src, m_image.size_in_bytes);
+    m_image.size_in_bytes = bytes_used;
```

Comment:
@hilookas why are you setting `size_in_bytes` directly here? that value should be set well before we get to this function. Also the `memcpy` line just below this one doesnt "know" about the `size_in_bytes` setting here at all for the image_t struct.

Reply:
That's because the length of the mjpeg encoded image fetched from v4l2 driver is changed between different frames.

We need record current length of the frame, and during the topic sending process, we need use the corrent length of the frame. Incorrect length will cause the memory leaking issue.

Hence, memcpy should use bytes_used (which is from v4l2 driver) rather than m_image.size_in_bytes